splink/blocking.py

```diff
@@ -48,6 +53,60 @@ def salted_blocking_rules(self):
             for n in range(self.salting_partitions):
                 yield f"{self.blocking_rule} and ceiling(l.__splink_salt * {self.salting_partitions}) = {n+1}"  # noqa: E501
 
+    @property
+    def _parsed_join_condition(self):
+        br = self.blocking_rule
+        return parse_one("INNER JOIN r", into=Join).on(
+            br, dialect=self.sqlglot_dialect
+        )  # using sqlglot==11.4.1
+
+    @property
+    def _join_conditions(self):
+        """
+        Extract the join conditions from the blocking rule as a tuple:
+        source_keys, join_keys, condition_expr
+
+        Returns:
+            list of tuples like [(name, name), (substr(name,1,2), substr(name,2,3))]
+        """
+
+        def remove_table_prefix(tree):
+            for c in tree.find_all(Column):
+                del c.args["table"]
```

Comment:
Just a general Q - is this better than using:
`c.args["table"] = None` here?

Reply:
Hmm, I'm not sure.  I guess ideally we'd want to be consistent with the rest of the Sqlglot codebase

Reply:
```
sqlglot.parse_one("l.first_name").args
```
results in 
```
{'this': (IDENTIFIER this: first_name, quoted: False),
 'table': (IDENTIFIER this: l, quoted: False),
 'db': None,
 'catalog': None}
```

```
sqlglot.parse_one("first_name").args
```
results in 
```
{'this': (IDENTIFIER this: first_name, quoted: False)}
```

i.e. there is no table key, meaning that `c.args["table"]` reflects how sqlglot would represent column without a table prefix